torch/utils/data/dataset.py

```diff
@@ -259,6 +259,46 @@ def __len__(self):
         return len(self.indices)
 
 
+class ChunkDataset(IterableDataset):
+    r"""
+    Wrapper class for the C++ ChunkDataset class.
```

Comment:
This is an implementation detail. Is it limited to that kind of dataset only? If yes, why aren't we checking that the user hasn't provided us with an object we don't expect to see?

Reply:
Yes, ChunkDataset is implemented based on `IterableDataset` because the size of the dataset is not known

We cant do any type checking because the C++ ChunkDataset API is based on templating (as much of Pytorch C++), which makes impossible to check if a specific ChunkDataset (e.g. DummyChunkDataset) is an instance of generic (templatized) `ChunkDataset ` class